src/radix_connect/p2p_links/p2p_link.rs

```diff
@@ -25,6 +27,10 @@ pub struct P2PLink {
     /// for the `ID`.
     pub connection_password: RadixConnectPassword,
 
+    pub connection_purpose: RadixConnectPurpose,
+
+    pub public_key: Ed25519PublicKey,
```

Comment:
doc... whos public key is this? Is this the public key of the Keyexchanged key? etc